metrics.go

```diff
@@ -99,3 +99,58 @@ func withHTTPMetrics(handler http.Handler, handlerName string) http.Handler {
 
 	return handler
 }
+
+var peersTotalMetric = prometheus.NewDesc(
+	prometheus.BuildFQName("ipfs", "p2p", "peers_total"),
+	"Number of connected peers",
+	[]string{"transport"},
+	nil,
+)
+
+// IpfsNodeCollector collects peer metrics
+type IpfsNodeCollector struct {
+	Node *Node
+}
+
+func (IpfsNodeCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- peersTotalMetric
+}
+
+func (c IpfsNodeCollector) Collect(ch chan<- prometheus.Metric) {
+	for tr, val := range c.PeersTotalValues() {
+		ch <- prometheus.MustNewConstMetric(
+			peersTotalMetric,
+			prometheus.GaugeValue,
+			val,
+			tr,
+		)
+	}
+}
+
+func (c IpfsNodeCollector) PeersTotalValues() map[string]float64 {
+	vals := make(map[string]float64)
+	if c.Node.host == nil {
+		return vals
+	}
+	for _, peerID := range c.Node.host.Network().Peers() {
```

Comment:
Do we only want the data transfer peers or, also the DHT peers. Since they are (optionally) on different hosts we may need to check there as well. If both do we want to deduplicate by peerID?

Reply:
@aschmahmann good point.

Now, I would like to compare perf between using a single host and using two hosts. I'm not sure it makes a difference tbh. Would you be ok with using a single host  for everything if thunderdome shows it is not worse?

Reply:
I'm fine with one host if behavior looks the same (i.e. if it's less complexity for a similar result).

IIUC this metric just tracks the peers we're connected to, not other things like bandwidth. Not sure how those are currently set up given two hosts.